src/sentry/eventstream/kafka/backend.py

```diff
@@ -272,8 +343,13 @@ def commit_offsets():
                 if error is not None:
                     raise Exception(error)
 
+                key = (message.topic(), message.partition())
+                if key not in owned_partition_offsets:
+                    logger.warning('Skipping message for unowned partition: %r', key)
```

Comment:
This is an error to me, in that if it happened I'd be worried about the state of the world. Unless I'm missing something?

Reply:
This was me being careful because I don't know how librdkafka manages messages that were already in its buffer to be delivered when a partition is revoked. I'm happy to upgrade this to an error if you're pretty confident it's not going to do that.